src/IB/BrinkmanPenalizationRigidBodyDynamics.cpp

```diff
@@ -74,56 +76,9 @@ set_rotation_matrix(const Eigen::Vector3d& rot_vel,
 inline Eigen::Vector3d
 solve_3x3_system(const Eigen::Vector3d& L, const Eigen::Matrix3d& T)
 {
-    const double a1 = T(0, 0);
-    const double a2 = T(0, 1);
-    const double a3 = T(0, 2);
-    const double b1 = T(1, 0);
-    const double b2 = T(1, 1);
-    const double b3 = T(1, 2);
-    const double c1 = T(2, 0);
-    const double c2 = T(2, 1);
-    const double c3 = T(2, 2);
-
-    const double d1 = L[0];
-    const double d2 = L[1];
-    const double d3 = L[2];
-
-    const double dnr = (a3 * b2 * c1 - a2 * b3 * c1 - a3 * b1 * c2 + a1 * b3 * c2 + a2 * b1 * c3 - a1 * b2 * c3);
-
-    Eigen::Vector3d W;
-
-    W[0] = (b3 * c2 * d1 - b2 * c3 * d1 - a3 * c2 * d2 + a2 * c3 * d2 + a3 * b2 * d3 - a2 * b3 * d3) / dnr;
-    W[1] = -(b3 * c1 * d1 - b1 * c3 * d1 - a3 * c1 * d2 + a1 * c3 * d2 + a3 * b1 * d3 - a1 * b3 * d3) / dnr;
-    W[2] = (b2 * c1 * d1 - b1 * c2 * d1 - a2 * c1 * d2 + a1 * c2 * d2 + a2 * b1 * d3 - a1 * b2 * d3) / dnr;
-
-    return W;
+    Eigen::FullPivLU<Eigen::Matrix3d> lu(T);
+    return lu.solve(L);
```

Comment:
Don't you like my pen-and-paper solution? ;-)

Reply:
While your rendition of Cramer's rule was beautiful, I really would prefer to use a more stable algorithm.